controllers/pipelinerun_controller.go

```diff
@@ -0,0 +1,203 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/triggers/pkg/filter"
+	"github.com/shipwright-io/triggers/pkg/inventory"
+	"github.com/shipwright-io/triggers/test/stubs"
+
+	"github.com/go-logr/logr"
+	tknv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// PipelineRunReconciler reconciles PipelineRun objects that may have triggers configured to generate
+// a BuildRun based on the Pipeline state.
+type PipelineRunReconciler struct {
+	client.Client                 // kubernetes client
+	Scheme        *runtime.Scheme // shared scheme
+	Clock                         // local clock instance
+
+	buildInventory *inventory.Inventory // local build triggers database
+}
+
+//+kubebuilder:rbac:groups=shipwright.io,resources=builds,verbs=get;list;watch
+//+kubebuilder:rbac:groups=shipwright.io,resources=buildruns,verbs=create;get;list;update;watch
+//+kubebuilder:rbac:groups=tekton.dev,resources=pipelineruns,verbs=get;list;update;watch
+
+// createBuildRun handles the actual BuildRun creation, uses the informed PipelineRun instance to
+// establish ownership. Only returns the created object name, and error.
+func (r *PipelineRunReconciler) createBuildRun(
+	ctx context.Context,
+	pipelineRun *tknv1beta1.PipelineRun,
+	buildName string,
+) (string, error) {
+	br := v1alpha1.BuildRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    pipelineRun.GetNamespace(),
+			GenerateName: fmt.Sprintf("%s-", pipelineRun.GetName()),
```

Comment:
My personal opinion here, I tend to think that we should take the Build name here. But maybe we need to go through the scenario to decide what is better. And maybe we also come to the conclusion that we want this to be specified in the Build's trigger definitions.

```suggestion
			GenerateName: fmt.Sprintf("%s-", buildName),
```

Reply:
I think it makes sense, Sascha. Taking in consideration trigger rules are defined in the `Build` object itself, it makes sense to follow its name. I'll make the change 👍